### PR TITLE
stats: auto analyze on certain period of a day

### LINKED ISSUE #7072: Make auto-analyze executes on certain hours in a day.
Some application doesn't want auto-analyze to be executed in the hours when the cluster is busy, we need to provide an option to only execute auto-analyze in idle hours.

We also need to consider that the stats of small tables may become out-of-date quickly, we should disregard the option and analyze them in time.

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 
Some application doesn't want auto-analyze to be executed in the hours when the cluster is busy, we need to provide an option to only execute auto-analyze in idle hours.
Fix #7072 

### What is changed and how it works?

Add two global variables to control the analyze start time and end time. But when the table is never analyzed, the time limit is ignored.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - None

Related changes

 - Need to update the documentation

PTAL @coocood @zz-jason @winoros

### CODE REVIEW

**Comment by winoros** [comment]
Can we let it auto analyze at `11:00pm~6:00am` by set this two variable?

**Comment by lamxTyler** [comment]
@winoros  Yes, you can set it to `23:00 CST` and `06:00 CST` like the case in test.

**Comment by shenli** [comment]
Could we guarantee that the EndTime is latter than the StartTime?

**Comment by coocood** [comment]
@lamxTyler 
I think using the local timezone is easier to use, and we don't need to specify the timezone part.

**Comment by lamxTyler** [comment]
@shenli Then it would need another field to note that they are not on the same day, I think it is more complicated.

**Comment by winoros** [comment]
Whether analyze it or not is still controlled by the existing variable `TiDBAutoAnalyeRatio`?

**Comment by coocood** [comment]
@lamxTyler 
The time zone of stats owner doesn't matter, we use the local timezone on the server that executes the set statement.
We can use UTC internally, but accept user input in local timezone.

File: statistics/update.go [hunk 1]
```diff
@@ -601,12 +601,28 @@ func TableAnalyzed(tbl *Table) bool {
...
+	if end.Sub(start) >= 0 {
+		return now.Sub(start) >= 0 && now.Sub(end) <= 0
+	}
+	// for cases like from 22:00 to 06:00
+	return now.Sub(end) <= 0 || now.Sub(start) >= 0
+}
+
+// NeedAnalyzeTable checks if we need to analyze the table:
 // 1. If the table has never been analyzed, we need to analyze it when it has
 //    not been modified for a time.
```
Comment (zhexuany) [inline statistics/update.go]: s/ not been modified for a time./ not been modified for a while

File: statistics/update_test.go [hunk 2]
```diff
@@ -918,3 +919,107 @@ func (s *testStatsUpdateSuite) TestLogDetailedInfo(c *C) {
...
+		{
+			tbl:    &statistics.Table{HistColl: statistics.HistColl{Columns: columns, ModifyCount: 1, Count: 1}},
+			limit:  0,
+			ratio:  0.3,
+			start:  "00:00 CST",
+			end:    "00:01 CST",
+			now:    "00:02 CST",
+			result: false,
+		},
+		// table already analyzed and but not within time period
```
Comment (zhexuany) [inline statistics/update_test.go]: s/table already analyzed and but not within time period/table is already analyzed and but not within time period

and for the rest of the comment, you also need fix them.

**Comment by lamxTyler** [comment]
@coocood How can the stats owner know the timezone of the server that sets the variable?

File: sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -623,6 +623,8 @@ var defaultSysVars = []*SysVar{
 	{ScopeSession, TiDBOptAggPushDown, boolToIntStr(DefOptAggPushDown)},
 	{ScopeGlobal | ScopeSession, TiDBBuildStatsConcurrency, strconv.Itoa(DefBuildStatsConcurrency)},
 	{ScopeGlobal, TiDBAutoAnalyzeRatio, strconv.FormatFloat(DefAutoAnalyzeRatio, 'f', -1, 64)},
+	{ScopeGlobal, TiDBAutoAnalyzeStartTime, DefAutoAnalyzeStartTime},
```
Comment (zz-jason) [inline sessionctx/variable/sysvar.go]: To safely use this feature, all the tidb-servers in a same cluster is required to be set in the same time zone?

Comment (zz-jason) [reply sessionctx/variable/sysvar.go]: never mind, the only thing that the DBAs need to do is to guarantee that the all the tidb-servers in the cluster can be used to analyze tables in the given time period based on the given time zone.

**Comment by coocood** [comment]
LGTM

**Pull request merged**